### PR TITLE
Basin allocation output

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/1122.

# Upgrade information

The flow rate in `flow.arrow` is now in a column named `flow_rate` rather than `flow` to be consistent with the input.

### CODE REVIEW

File: core/src/allocation_optim.jl [hunk 1]
```diff
@@ -475,7 +459,76 @@ function adjust_basin_capacities!(
...
+                if priority_idx == basin_priority_idx
+                    # Basin demand
+                    _, basin_idx = id_index(basin.node_id, node_id)
+                    demand += basin.demand[basin_idx]
+                end
+                allocated =
+                    JuMP.value(F_basin_in[node_id]) - JuMP.value(F_basin_out[node_id])
+                # TODO: Abstracted for a basin is not so clear, maybe it should be Δstorage/Δt
+                # over the last allocation interval?
+                abstracted = 0.0
```
Comment (visr) [inline core/src/allocation_optim.jl:508]: I think `Δstorage/Δt` would make sense. Could you create an issue so we can track it?

Also, `abstracted` made a lot of sense when we were only talking about Users. Now we added water level control and soon we should add instream demand, both of which don't **abstract** water. Shall we already rename the column to `realized`? Or maybe @gijsber has other suggestions, perhaps `flow_rate`?

Actually if you don't mind, I realize just now we consistently use `flow_rate` in input tables, but `flow` columns in `flow.arrow` and `allocation_flow.arrow`. Mind updating those as well?

File: core/src/allocation_optim.jl [hunk 2]
```diff
@@ -475,7 +459,76 @@ function adjust_basin_capacities!(
...
+            end
+        end
+
+        if has_demand
+            # Save allocations and demands to record
+            push!(record_demand.time, t)
+            push!(record_demand.subnetwork_id, allocation_network_id)
+            push!(record_demand.node_type, String(Symbol(node_id.type)))
```
Comment (visr) [inline core/src/allocation_optim.jl:516]:
```suggestion
            push!(record_demand.node_type, string(node_id.type))
```

**Review by visr — COMMENTED** [review]
Good stuff. Left some comments.

File: core/src/write.jl [hunk 3]
```diff
@@ -205,19 +207,19 @@ function allocation_flow_table(
...
-        record.edge_id,
-        record.from_node_id,
-        record.to_node_id,
-        record.subnetwork_id,
-        record.priority,
-        record.flow,
-        record.collect_demands,
+        record_flow.edge_id,
+        record_flow.from_node_id,
+        record_flow.to_node_id,
```
[the remark this answers is not in the document]
Comment (SouthEndMusic) [reply core/src/write.jl:218]: I think you mean `to_node_type` for the second suggestion?

File: core/src/allocation_optim.jl (hunk 1 above)
Comment (SouthEndMusic) [reply core/src/allocation_optim.jl:508]: Well, if we're breaking stuff anyway, might as well do a lot xd

File: core/src/write.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (visr) [reply core/src/write.jl]: Yes, sorry.

**Review by visr — APPROVED** [review]

**Pull request merged**